google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java

```diff
@@ -283,6 +284,11 @@ private void createUIComponents() {
   }
 
   private void initEvents() {
+    // track all changes in UI to report settings changes.
+    UserActivityWatcher activityWatcher = new UserActivityWatcher();
```

Comment:
this is cool!

Reply:
Yeah we can use it directly when we need something like that, pretty simple.